models/dataset.go

```diff
@@ -286,17 +287,19 @@ func CreateDataset(reader io.Reader) (*Dataset, error) {
 }
 
 // CreateVersion manages the creation of a version from a reader
-func CreateVersion(reader io.Reader) (*Version, error) {
+func CreateVersion(reader io.Reader, datasetID string) (*Version, error) {
 	b, err := ioutil.ReadAll(reader)
 	if err != nil {
 		return nil, errs.ErrUnableToReadMessage
 	}
 
+	var version Version
+	version.ID = datasetID
```

Comment:
This is not needed the, version id is set to a new uuid, see line 301.

As James mentioned in the title the version should return the dataset_id in the response body (version resource), see line 302.